src/runtime/getPositions.ts

```diff
@@ -238,32 +265,6 @@ async function resolveAppTokenPosition(
       })
     ]!
 
-  const appTokenContract = {
-    address: positionDefinition.address as Address,
-    abi: erc20Abi,
-  } as const
-  const contractData = await getClient(positionDefinition.networkId).multicall({
-    contracts: [
-      {
-        ...appTokenContract,
-        functionName: 'balanceOf',
-        args: [(address as Address) ?? zeroAddress], // TODO: this is incorrect for intermediary app tokens
-      },
-      {
-        ...appTokenContract,
-        functionName: 'totalSupply',
-      },
-    ],
-    allowFailure: false,
-  })
-
-  let balance = contractData[0]
-  const totalSupply = contractData[1]
-  // If no user address is provided, use 0
-  if (!address) {
-    balance = 0n
-  }
-
```

Comment:
This was moved out of this function, directly when fetching tokensInfo, this way it's much faster cause it benefits from the client batching.